ray-operator/controllers/ray/utils/util.go

```diff
@@ -445,7 +445,7 @@ func CalculateMinResources(cluster *rayv1.RayCluster) corev1.ResourceList {
 
 // CalculatePodResource returns the total resources of a pod.
 // Request values take precedence over limit values.
-func CalculatePodResource(podSpec corev1.PodSpec) corev1.ResourceList {
+func CalculatePodResource(podSpec corev1.PodSpec, numOfHosts int32) corev1.ResourceList {
```

Comment:
Passing `numOfHosts` into `CalculatePodResource` is an abstraction leak. Could you add a function `calculateReplicaResource(podSpec, numOfHosts)`?

```go
func calculateReplicaResource(podSpec, numOfHosts) {
  return CalculatePodResource(podSpec) * numOfHosts
}
```